sklearn/manifold/_spectral_embedding.py

```diff
@@ -367,7 +367,7 @@ def spectral_embedding(
             # We increase the number of eigenvectors requested, as lobpcg
             # doesn't behave well in low dimension and create initial
             # approximation X to eigenvectors
-            X = random_state.rand(laplacian.shape[0], n_components + 1)
+            X = random_state.randn(laplacian.shape[0], n_components + 1)
```

Comment:
In principle, could this change result in a different embedding with the same inputs?

Reply:
@thomasjpfan - thanks for checking!

I also do not see it specifically mentioned in the papers - common sense suggest that vectors with only positive components cover the positive quadrant only, not the whole vector space, are not so good to approximate eigenvectors. If a reference is required, all example in the original MATLAB/Octave code, written for the paper,  https://github.com/lobpcg/blopex/blob/master/blopex_tools/matlab/lobpcg/lobpcg.m use randn, e.g., line 97. 

Changing the initial approximation for the solver may result in different approximate eigenvectors and thus a different embedding, or may make the solver to fail, but both situations are normally rare if the number of iterations executed is not too small.

Reply:
If the embedding can change, then I think we need to add the models that are affected in the `Changed models` section:

https://github.com/scikit-learn/scikit-learn/blob/e707e61d85127dae1f58e9ad2689ca2708add0a4/doc/whats_new/v1.1.rst?plain=1#L23-L29

Reply:
Changing the initial approximation for the solver may result in different approximate eigenvectors just because it is random,  if the number of iterations executed is too small.. It is random in the current version, so if one simply runs it with a different seed that may in principle affect the embedding. It is a fact unrelated to this PR.  
 
However, the number of iterations executed here is not too small by default, so the embedding is unlikely going to be changed in practice. I am unsure how the models and which models could be affected to this PR, which simply changes one random distribution with another.

Reply:
I have noticed that  the components of the initial vectors in the docstring example https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.lobpcg.html?highlight=random also uses uniformly distributed on the interval [0, 1], so changing it to normal around zero in 
https://github.com/scipy/scipy/pull/15154

Reply:
> It is random in the current version, so if one simply runs it with a different seed

Lets say a user fixes `random_state=0`, one expects that the embedding they get in `scikit-learn==1.0` is the same as `scikit-learn==1.1`. I can see that the embedding does not change in practice. On the other hand, if there is a chance that the embedding changes with the same parameters, then I think we should be transparent and inform our users.

> I am unsure how the models and which models could be affected to this PR,

I think it's only `SpectralEmbedding` and `SpectralClustering`.

Reply:
Got it, thanks for your patience! I'll do it now.